### ISSUE TITLE
When a subscription is upgraded or downgraded, the purchaseToken does not change

### ISSUE THREAD

**Issue by sceneren** [mvojtkovszky/BillingHelper#11]
```
mBilling.launchPurchaseFlow(
                activity = ActivityUtils.getTopActivity(),
                productName = realPriceInfo.code,
                obfuscatedAccountId = orderSn,
                subscriptionParams = SubscriptionPurchaseParams(
                    basePlanId = realPriceInfo.basePlanId,
                    offerId = realPriceInfo.offerId,
                    updateOldToken = mBilling.getPurchaseWithProductName(realPriceInfo.code)?.purchaseToken,
//                    updateExternalTransactionId = oldPurchase?.orderId,
                    updateReplacementMode = BillingFlowParams.SubscriptionUpdateParams.ReplacementMode.CHARGE_FULL_PRICE
                )
            )
```
When I complete the payment using the above code, After I use the above code to complete the payment, and get the latest purchaseToken through ```mBilling.getPurchaseWithProductName(realPriceInfo.code)```, I find that it has not changed

**Comment by darkokoa** [comment]
I encountered the same issue. My temporary solution was to access the `purchases` field in `BillingHelper` through reflection, and `purchases.findLast { it.products.contains(productName) }` is the latest purchase you want.

**Comment by darkokoa** [comment]
Hi @mvojtkovszky, could you add a function to get the latest purchase? Or is there any other flexible way to access the purchases?

**Comment by mvojtkovszky** [comment]
Thanks for the tips @sceneren and @darkokoa . I addressed this with latest [develop commit](https://github.com/mvojtkovszky/BillingHelper/commit/033b3ff5048f16b716d1b9a3491c476241498a40).
So try building with 033b3ff504 [as the version](com.github.mvojtkovszky:BillingHelper:033b3ff504).

Will pick it up with next release if that works as expected.

**Comment by sceneren** [comment]
Thanks for your help, it works fine in version 033b3ff

**Comment by mvojtkovszky** [comment]
Fixed with [3.4.0](https://github.com/mvojtkovszky/BillingHelper/releases/tag/3.4.0)

renamed `getPurchaseWithProductName` with `getPurchasesWithProductName` that now returns a list of products instead of first found.
